src/renderer/binary.ts

```diff
@@ -2,14 +2,40 @@ import * as fs from 'fs-extra';
 import * as path from 'path';
 import extract from 'extract-zip';
 
-import { VersionState } from '../interfaces';
+import { Version, VersionSource, VersionState } from '../interfaces';
 import { normalizeVersion } from '../utils/normalize-version';
 import { USER_DATA_PATH } from './constants';
 import { removeTypeDefsForVersion } from './fetch-types';
 import { AppState } from './state';
 import { download as electronDownload } from '@electron/get';
 
-const pendingDownloads: Record<string, Promise<void>> = {};
+// versions that are currently being downloaded
+const downloading: Map<string, Promise<void>> = new Map();
+
+// versions that are currently being unzipped
+const unzipping: Set<string> = new Set();
+
+/**
+ * Determine by inspection the VersionState of this version
+ *
+ * @param {Version} version
+ * @returns {VersionState} the state of the version
+ */
+export function getVersionState(ver: Version): VersionState {
+  const { localPath, version } = ver;
+
+  if (downloading.has(version)) {
+    return VersionState.downloading;
+  }
+
+  if (unzipping.has(version)) {
+    return VersionState.unzipping;
+  }
+
+  const dir = localPath || getDownloadPath(version);
+  const exec = path.join(dir, execSubpath());
```

Comment:
That's a perceptive question!

Yes there is: the current semantics of `getElectronBinaryPath()` "the first parameter is ignored if you pass in a second parameter" are weird, especially since both of the places we call this function in production pass two parameters. So this is a step towards refactoring `getElectronBinaryPath()` in a followup PR.